cli/file_watcher.rs

```diff
@@ -15,57 +13,42 @@ use notify::Error as NotifyError;
 use notify::RecommendedWatcher;
 use notify::RecursiveMode;
 use notify::Watcher;
-use pin_project::pin_project;
 use std::collections::HashSet;
 use std::path::PathBuf;
-use std::pin::Pin;
 use std::sync::Arc;
-use std::task::Context;
-use std::task::Poll;
 use std::time::Duration;
-use tokio::pin;
 use tokio::select;
+use tokio::sync::mpsc;
 use tokio::time::sleep;
-use tokio::time::Instant;
-use tokio::time::Sleep;
 
 const DEBOUNCE_INTERVAL: Duration = Duration::from_millis(200);
 
-#[pin_project(project = DebounceProjection)]
-struct Debounce {
-  #[pin]
-  timer: Sleep,
-  changed_paths: Arc<Mutex<HashSet<PathBuf>>>,
+struct DebouncedReceiver {
```

Comment:
I've just lazily made this very implementation specific because we don't seem to have a folder where we can just throw in these kind of utility structs or I'm not sure of a specific one to put it in. Perhaps it would be useful to extract this out in the future.